app/resources/applications/ApplicationsController.jl

```diff
@@ -497,10 +505,22 @@ function pages(app)
 end
 
 function available_port()
-  replport = rand(50_000:60_000)
-  isempty(find(Application, replport = replport)) || available_port()
+  isempty(SearchLight.find(Application)) && return (first(PORTS_RANGE), first(PORTS_RANGE)+1)
+  usedports = [app.port for app in SearchLight.find(Application)]
+  last(usedports)+1 >= last(PORTS_RANGE) && throw("$PORTS_RANGE ports are all in use, delete some apps")
```

Comment:
This won't work if we free ports at the beginning of the range. 
Also `last` won't work if the query does not sort by port id (should be `max`).